atom/browser/api/atom_api_web_contents.h

```diff
@@ -133,6 +133,10 @@ class WebContents : public mate::TrackableObject<WebContents>,
   void SetAllowTransparency(bool allow);
   bool IsGuest() const;
 
+  // Handler for permission requests.
+  void SetPermissionRequestHandler(v8::Local<v8::Value> val,
```

Comment:
Sounds good to me to remove `permission-request` event if we have `session.setPermissionRequestHandler` API.